client/state/sites/selectors/get-wpcom-follower-role.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import { getSite } from 'calypso/state/sites/selectors';
+
+/**
+ * Retrieve the WPCOM follower role info, based on site settings
+ *
+ * @param  {object} state    Global state tree
+ * @param  {number} siteId   Site ID
+ * @returns {object}         Follower role object
+ */
+const getWpcomFollowerRole = ( state, siteId ) => {
+	const site = getSite( state, siteId );
+	const displayName = site.is_private
+		? translate( 'Viewer', { context: 'Role that is displayed in a select' } )
+		: translate( 'Follower', { context: 'Role that is displayed in a select' } );
+
+	return {
+		display_name: displayName,
+		name: 'follower',
+	};
+};
```

Comment:
I think this function shouldn't be a selector at all, as it depends on an external `translate` function (a proper selector should depend only on `state` and nothing else) and doesn't react reliably to changes in locale. Function like this would work better:
```js
function getWpcomFollowerRole( siteIsPrivate, translate ) { ... }
```
Let the calling component supply the arguments by calling the `isPrivateSite` selector and getting `translate` from `localize` or `useTranslate`.

Reply:
Agreed although I think it's out of scope for this PR to change that. Happy to follow up with another PR right after this one if that works for you?

Reply:
Yes, let's do a follow-up :+1: